### PR TITLE
fix(VertexCycle): Added a version of the VertexCycle which keeps the …

### PR DESCRIPTION
**Pull request by MelchiorSchuh**
…orientation of the vertices.

### CODE REVIEW

File: tests/mesh/test-vertex-cycle.cpp [hunk 1]
```diff
@@ -66,9 +66,42 @@ void test_vertex_cycle()
...
+    const geode::detail::OrientedVertexCycle< std::array< geode::index_t, 2 > >
+        cycle7{ { 3, 1 } };
+
+    OPENGEODE_EXCEPTION( cycle1 == cycle2,
+        "[Test] Wrong result for operator== with cycle1 and cycle2" );
+    OPENGEODE_EXCEPTION( cycle1 != cycle3,
+        "[Test] Wrong result for operator== with cycle1 and cycle3" );
+    OPENGEODE_EXCEPTION( cycle1 != cycle4,
+        "[Test] Wrong result for operator== with cycle1 and cycle4" );
+    OPENGEODE_EXCEPTION( cycle1.is_opposite( cycle4 ),
```
Comment (panquez) [inline tests/mesh/test-vertex-cycle.cpp:93]: check message

File: include/geode/mesh/core/detail/vertex_cycle.h [hunk 2]
```diff
@@ -74,6 +74,22 @@ namespace geode
...
+            }
+            polygon_sort( vertices );
+        }
+
+        template <>
+        inline void oriented_rotate( std::array< index_t, 2 >& vertices )
+        {
+            rotate( vertices );
+        }
+
```
Comment (panquez) [inline include/geode/mesh/core/detail/vertex_cycle.h:92]: is it different from l.68 ? should we have two implementations or use the same?

**Review by panquez — APPROVED** [review]

File: include/geode/mesh/core/detail/vertex_cycle.h [hunk 3]
```diff
@@ -52,80 +52,147 @@ namespace geode
...
 
+        template < typename Container >
+        inline void rotate( Container& vertices )
+        {
+            oriented_rotate( vertices );
+            remove_orientation( vertices );
+        }
+
         template <>
         inline void rotate( std::array< index_t, 2 >& vertices )
```
Comment (BotellaA) [inline include/geode/mesh/core/detail/vertex_cycle.h:90]: Is `rotate` still used?

**Review by BotellaA — APPROVED** [review]

File: include/geode/mesh/core/detail/vertex_cycle.h (hunk 3 above)
Comment (MelchiorSchuh) [reply include/geode/mesh/core/detail/vertex_cycle.h:90]: No, but wouldn't removing it be a breaking change ?

**Pull request merged**